preprocessing/science/refine_srf.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+import pyproj
+import matplotlib.pyplot as plt
+import scipy.ndimage
+from scipy import interpolate
+from math import floor
+import os
+
+
+class FaultPlane:
+    def __init__(self, nx, ny):
+        self.nx = nx
+        self.ny = ny
+        self.ndt = 0
+        self.lon = np.zeros((ny, nx))
+        self.lat = np.zeros((ny, nx))
+        self.x = np.zeros((ny, nx))
+        self.y = np.zeros((ny, nx))
+        self.depth = np.zeros((ny, nx))
+        self.t0 = np.zeros((ny, nx))
+        self.slip1 = np.zeros((ny, nx))
+        self.strike = np.zeros((ny, nx))
+        self.dip = np.zeros((ny, nx))
+        self.rake = np.zeros((ny, nx))
+        self.aSR = 0
+        self.PSarea = 0
+        self.dt = 0
+        self.myt = 0
+
+    def init_aSR(self):
+        self.aSR = np.zeros((self.ny, self.nx, self.ndt))
```

Comment:
Only one sliprate, but according to srf documentation it is possible, that we have also three slip rates per subfault.